### PR TITLE
Kan 65 sportplatz anträge im backend implementieren

### PR DESCRIPTION
**Pull request by Ilimea**
Entity für Sportfeld Antrag
Controller
Service
Repository
Event
Eventhandler

→ Mediator Pattern verwenden

### CODE REVIEW

File: OlympGuide.Application/Features/SportFieldProposal/SportFieldProposalService.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            var sportFieldProposal = await _repository.ChangeStateById(sportFieldProposalId,newState);
+
+            if (sportFieldProposal == null)
+            {
+                throw new NoSportFieldFoundException(sportFieldProposalId);
+            }
+
+            if(newState == SportFieldProposalStates.Open)
+            {
+                var Event = new SportFieldProposalAcceptedEvent(sportFieldProposal);
```
Comment (SuffPanda) [inline OlympGuide.Application/Features/SportFieldProposal/SportFieldProposalService.cs]: lowercase

File: OlympGuide/Controllers/SportFieldProposalController.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        private readonly IMapper _mapper = mapper;
+        private readonly ISportFieldProposalService _service = service;
+
+        [HttpGet("")]
+        [Authorize("access:admin")]
+        public async Task<List<SportFieldProposalDetailsDto>> GetAllSportFieldProposals([FromQuery] SportFieldProposalStates state)
+        {
+            var list = await _service.GetAllSportFieldProposals();
+            if (state.Equals(SportFieldProposalStates.Open))
```
Comment (SuffPanda) [inline OlympGuide/Controllers/SportFieldProposalController.cs]: rewrite to if open then getAllOpen... else getAll..

File: OlympGuide.Application/Features/SportField/SportFieldValidation.cs [hunk 3]
```diff
@@ -1,36 +1,32 @@
-﻿namespace OlympGuide.Application.Features.SportField
+﻿using OlympGuide.Domain.Features.SportField;
+
+namespace OlympGuide.Application.Features.SportField
```
Comment (mueld) [inline OlympGuide.Application/Features/SportField/SportFieldValidation.cs]: As this class validates a domain entity, it should be moved to the domain layer. It contains specific domain logic.
Don't forget to adapt the namespace when you move the file!

File: OlympGuide.Application/Features/SportFieldProposal/ISportFieldProposalService.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+{
+    public interface ISportFieldProposalService
+    {
+        public Task<List<SportFieldProposalType>> GetAllSportFieldProposals();
+        public Task<List<SportFieldProposalType>> GetAllOpenSportFieldProposals();
+        public Task<SportFieldProposalType> GetSportFieldProposalById(Guid sportFieldProposalId);
+        public Task<SportFieldProposalType> AddSportFieldProposal(SportFieldProposalDto sportFieldProposalToAdd);
+        public Task<SportFieldProposalType> ChangeStateById(Guid sportFieldProposalId, SportFieldProposalStates newState);
+    }
+}
```
Comment (mueld) [inline OlympGuide.Application/Features/SportFieldProposal/ISportFieldProposalService.cs:14]: Interface belongs to the domain layer

File: OlympGuide.Application/Features/SportFieldProposal/NoSportFieldProposalFoundException.cs [hunk 5]
```diff
@@ -0,0 +1,4 @@
+﻿namespace OlympGuide.Application.Features.SportFieldProposal
+{
+    public class NoSportFieldProposalFoundException(Guid id) : Exception(String.Format("Sport field proposal with id: {0} was not found", id.ToString()));
+}
```
Comment (mueld) [inline OlympGuide.Application/Features/SportFieldProposal/NoSportFieldProposalFoundException.cs]: belongs to the domain layer

File: OlympGuide.Application/Features/SportFieldProposal/SportFieldProposalApprovedEvent.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+namespace OlympGuide.Application.Features.SportFieldProposal
+{
+    public class SportFieldProposalApprovedEvent : INotification
+    {
+        public SportFieldProposalApprovedEvent(SportFieldProposalType sportFieldProposal) {
+            SportFieldProposal = sportFieldProposal;
+        }
+        public SportFieldProposalType SportFieldProposal { get; init; }
+    }
+}
```
Comment (mueld) [inline OlympGuide.Application/Features/SportFieldProposal/SportFieldProposalApprovedEvent.cs]: move to domain layer

File: OlympGuide.Infrastructre/Repositories/SportFieldProposalRepository.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+namespace OlympGuide.Infrastructre.Repositories
+{
+    public class SportFieldProposalRepository(OlympGuideDbContext context) : ISportFieldProposalRepository
+    {
+        public async Task<List<SportFieldProposalType>> GetAllSportFieldProposals()
+        {
+            var sportFieldProposalList = await context.SportFieldProposals
+                .ToListAsync();
+
+            return sportFieldProposalList;
```
Comment (mueld) [inline OlympGuide.Infrastructre/Repositories/SportFieldProposalRepository.cs]: `return  await context.SportFieldProposals .ToListAsync();`

File: OlympGuide.Infrastructre/Repositories/SportFieldProposalRepository.cs [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        public async Task<List<SportFieldProposalType>> GetAllOpenSportFieldProposals()
+        {
+            var sportFieldProposalList = await context.SportFieldProposals
+                .Where(sf => sf.State == SportFieldProposalStates.Open)
+                .ToListAsync();
+
+            return sportFieldProposalList;
```
Comment (mueld) [inline OlympGuide.Infrastructre/Repositories/SportFieldProposalRepository.cs]: same

File: OlympGuide.Infrastructre/Repositories/SportFieldProposalRepository.cs [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        public async Task<SportFieldProposalType> GetSportFieldProposalById(Guid sportFieldProposalId)
+        {
+            var sportFieldProposal = await context.SportFieldProposals
+                .SingleAsync(sf => sf.Id == sportFieldProposalId);
+
+            return sportFieldProposal;
```
Comment (mueld) [inline OlympGuide.Infrastructre/Repositories/SportFieldProposalRepository.cs]: same

**Review by mueld — CHANGES_REQUESTED** [review]

File: OlympGuide.Application/Features/SportFieldProposal/ISportFieldProposalService.cs (hunk 4 above)
Comment (Ilimea) [reply OlympGuide.Application/Features/SportFieldProposal/ISportFieldProposalService.cs:14]: nach Absprache in Application gelassen

**Review by mueld — APPROVED** [review]

**Pull request merged**